launcher/main.cpp

```diff
@@ -25,17 +26,27 @@
 #endif
 int argcForClient;
 char ** argvForClient;
+#elif defined(VCMI_ANDROID)
+#include <QAndroidJniObject>
+#include <QtAndroid>
 #endif
 
-int main(int argc, char * argv[])
+// android must export main explicitly to make it visible in the shared library
+#ifdef VCMI_ANDROID
+#define EXPORT ELF_VISIBILITY
+#else
+#define EXPORT
+#endif
+
+int EXPORT main(int argc, char * argv[])
 {
 	int result;
 #ifdef VCMI_IOS
 	{
 #endif
 	QApplication vcmilauncher(argc, argv);
```

Comment:
```suggestion
	QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);

	QApplication vcmilauncher(argc, argv);
```

Can you try to set this attribute. On my device (Samsung S23) it looks much better afterwards. Maybe we also have to shorten some texts.